mtags/src/main/scala-3/scala/meta/internal/pc/Signatures.scala

```diff
@@ -0,0 +1,269 @@
+import dotty.tools.dotc.core.Names.Name
+import dotty.tools.dotc.core.NameKinds.EvidenceParamName
+
+import dotty.tools.dotc.core.Flags._
+
+trait Signatures {
+
+  class ShortenedNames(context: Context) {
+    val history = collection.mutable.Map.empty[Name, ShortName]
+
+    def lookupSymbol(short: ShortName): Type = {
+        import scala.meta.internal.mtags.MtagsEnrichments._
+        context.findRef(short.name)
+    }
+
+    def tryShortenName(short: ShortName)(using Context): Boolean = {
+      history.get(short.name) match {
+        case Some(ShortName(_, other)) => true
+        case None =>
+          val foundTpe = lookupSymbol(short)
+          val syms = List(foundTpe.termSymbol, foundTpe.typeSymbol)
+          val isOk = syms.filter(_ != NoSymbol) match {
+            case Nil => false
+            case founds => founds.exists(_ == short.symbol)
+          }
+          if (isOk) {
+            history(short.name) = short
+            true
+          } else {
+            false
+          }
+      }
+    }
+
+    def tryShortenName(name: Option[ShortName])(using Context): Boolean =
+      name match {
+        case Some(short) => tryShortenName(short)
+        case None => false
+      }
+  }
+
+  /** Compute detail string for CompletionItem#detail, which is also used by
+    * method's signature string in label.
+    * 
+    * for
+    * - method: method signature
+    *   - e.g. `[A: Ordering](x: List[Int]): A`
+    * - class or module: package name that class or module is belonging
+    *   - e.g. ` java.lang`
+    * - otherwise: its type
+    *   - e.g. ` java.lang.String` ` Symbols.Symbol`
+    */
+  def infoString(sym: Symbol, info: Type, history: ShortenedNames)(using Context): String = {
+    sym match {
+      case m if m.is(Method) =>
+        SignaturePrinter(sym, history, info).defaultMethodSignature()
+      case _ if sym.isClass || sym.is(Module) =>
+        val printer = SymbolPrinter()
+        s" ${printer.fullNameString(sym.owner)}"
+      case _ =>
+        val short = shortType(info, history)
+        val printer = SymbolPrinter()
+        s" ${printer.typeString(short)}"
+    }
+  }
+
+  class SignaturePrinter(
+      gsym: Symbol,
+      shortenedNames: ShortenedNames,
+      gtpe: Type,
+      // includeDocs: Boolean,
+      // includeDefaultParam: Boolean = true,
+      // printLongType: Boolean = true
+  )(using Context) {
+    // In case rawParamss is no set, fallback to paramSymss
+    private val paramss = if (gsym.rawParamss.length != 0) gsym.rawParamss else gsym.paramSymss
+    private lazy val implicitParams: List[Symbol] = paramss.flatMap(params => params.filter(p => p.is(Implicit)))
+
+    // should be able to filter by `p.name.is(EvidenceParamName)` or `p.name.startsWith(EvidenceParamName.separator)`
+    // but it sometimes doesn't work (not sure why), therefore workaround by string comparison
+    private lazy val implicitEvidenceParams: Set[Symbol] =
```

Comment:
I'm not yet sure exactly when this happens, I'll try to create a minimum reproducible example.

Reply:
I realized that `NameKind` properties exist only in the in-project names.
This NameKind is generated from https://github.com/tanishiking/dotty/blob/65b17af639e17a5d526581ff68fc17bfa08e6299/compiler/src/dotty/tools/dotc/ast/Desugar.scala#L190-L195 (, and I guess 3rd-party library doesn't deliver the NameKind information because Tasty doesn't have NameKind information).